Source/WebCore/html/HTMLAttachmentElement.cpp

```diff
@@ -121,6 +124,32 @@ static const AtomString& attachmentSubtitleIdentifier()
     return identifier;
 }
 
+static const AtomString& attachmentSaveIdentifier()
+{
+    static MainThreadNeverDestroyed<const AtomString> identifier("attachment-save"_s);
+    return identifier;
+}
+
+static QualifiedName saveQualifiedName()
+{
+    return QualifiedName { nullAtom(), "save"_s, nullAtom() };
+}
+
+void HTMLAttachmentElement::addShadowElement(RefPtr<HTMLElement> element, const AtomString& id, String innerText)
```

Comment:
Can we just combine `createShadowDivElement` and `addShadowElement` ?

`addShadowElement` is an odd function to call on its own.

Reply:
I needed `addShadowElement` for adding the inner attachment, because it's constructed in a special way that `createShadowDivElement` cannot do.

Reply:
My point is that it feels odd even when building inner attachment, I'd rather have a bit more duplication here.